### PR TITLE
Merge arguments from supervisor into endpoint configuration

### PR DESCRIPTION
**Pull request by hauleth**
This allows passing additional configuration options in supervisor spec, ex.:

```elixir
children = [
  {MyApp.Endpoint, url: [path: "/api"]}
]
```

Will be the same as setting:

```elixir
config :my_app, MyApp.Endpoint, url: [path: "/api"]
```

---

Configuration still overwrite values set in supervisor, so:

```elixir
children = [
  {MyApp.Endpoint, url: [path: "/foo"]}
]
```

and

```elixir
config :my_app, MyApp.Endpoint, url: [path: "/bar"]
```

Will result in:

```elixir
assert "/bar" == MyApp.Endpoint.config(:url)[:path]
```

### CODE REVIEW

File: lib/phoenix/config.ex [hunk 1]
```diff
@@ -70,9 +70,14 @@ defmodule Phoenix.Config do
   Useful to read a particular value at compilation time.
   """
   def from_env(otp_app, module, defaults) do
-    Keyword.merge(defaults, fetch_config(otp_app, module), &merger/3)
+    otp_app
+    |> fetch_config(module)
+    |> merge(defaults)
   end
 
+  @doc false
```
Comment (josevalim) [inline lib/phoenix/config.ex]: Please document it accordingly. :) Even if used only internally.

File: lib/phoenix/config.ex [hunk 2]
```diff
@@ -70,9 +70,14 @@ defmodule Phoenix.Config do
   Useful to read a particular value at compilation time.
   """
   def from_env(otp_app, module, defaults) do
-    Keyword.merge(defaults, fetch_config(otp_app, module), &merger/3)
+    otp_app
+    |> fetch_config(module)
+    |> merge(defaults)
```
Comment (josevalim) [inline lib/phoenix/config.ex]: Reversing the order of merge but calling it merge is potentially very confusing. Or we keep the order and swap arguments here (non-pipeable) or we call it something like `reverse_merge`. I prefer the former.

Comment (hauleth) [reply lib/phoenix/config.ex]: Maybe name it `override` or `into_defaults`.

File: lib/phoenix/config.ex [hunk 3]
```diff
@@ -72,11 +72,16 @@ defmodule Phoenix.Config do
...
     |> fetch_config(module)
-    |> merge(defaults)
+    |> override(defaults)
   end
 
-  @doc false
-  def merge(a, b), do: Keyword.merge(b, a, &merger/3)
+  @doc """
+  Take 2 keyword lists and override values from first one with values from
+  the second one recursively.
```
Comment (josevalim) [inline lib/phoenix/config.ex]: You are describing merge here, and not overrides. I think this is a strong sign we should just go with merge please? Less chance of confusion. :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**